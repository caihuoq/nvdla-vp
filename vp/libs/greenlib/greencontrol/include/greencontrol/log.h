//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
//   This program is free software.
// 
//   If you have no applicable agreement with GreenSocs Ltd, this software
//   is licensed to you, and you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation; either version 2 of the License, or
//   (at your option) any later version.
// 
//   If you have a applicable agreement with GreenSocs Ltd, the terms of that
//   agreement prevail.
// 
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
// 
//   You should have received a copy of the GNU General Public License
//   along with this program; if not, write to the Free Software
//   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
//   02110-1301  USA 
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __LOG_H__
#define __LOG_H__

#include "greencontrol/gclog/gc_logger.h"
#include "greencontrol/gclog/filter.h"
#include "greencontrol/gclog/output.h"

#endif
