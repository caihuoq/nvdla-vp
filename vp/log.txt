-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for TLM2C
-- TLM2C include directory = /workspace/vp/libs/tlm2c/include
-- TLM2C library directory = /workspace/vp/libs/tlm2c.build
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SimpleCPU model.
-- SimpleCPU include directory = /workspace/vp/models/cpu/include
-- SimpleCPU library directory = /workspace/vp/models/cpu.build
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/vp
