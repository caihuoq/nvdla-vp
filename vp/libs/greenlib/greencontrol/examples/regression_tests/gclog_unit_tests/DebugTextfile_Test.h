//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
// The contents of this file are subject to the licensing terms specified
// in the file LICENSE. Please consult this file for restrictions and
// limitations that may apply.
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __DEBUGTEXTFILE_TEST_H__
#define __DEBUGTEXTFILE_TEST_H__

#include "greencontrol/log.h"
#include "TestCase.h"

// GCLOG_RETURN should be enabled
class DebugTextfile_Test : public TestCase
{
public:
  DebugTextfile_Test() {}
  virtual ~DebugTextfile_Test(void) {}

  std::string identify()
  {
    return "DebugTextfile_Test";
  }

  void setup()
  {
    pLog = new GC_Logger();
  }

  void destroy()
  {
    delete pLog;
  }

  // This is for testing the DebugTextfile class
  void run()
  {
    // Create DebugTextfile
    DebugTextfile textOut("DebugTextfile_Test.txt", false);
    ControlTransactionHandle ct, ct2;
    LogConfig cfg;

    ct = getCT();
    ct2 = getCT();
    ct->resetChangeFlag();

    // Test 1: The following outputs should be printed in the DebugTextfile_Test.txt
    textOut.write("This should be the first line in DebugTextfile_Test.txt after the header.");
    textOut.writePure("This should be the second line, but without the \"User output:\" prefix.");

    // Test 2: An unmodified transaction should be printed, with all debug info.
    textOut.writePure("\nTest 2\nAfter this, there should be a single transaction.");
    textOut.processTransaction(ct);

    // Test 3: Two transactions should be printed. One was modified, so it should act as a returning transaction.
    textOut.writePure("\nTest 3\nNow the same transaction should be printed, followed by another, which acts as a returning CT");
    ct2->set_mService(AV_SERVICE);
    ct2->set_mSpecifier("This transaction should be modified.");
    ct2->set_mMetaData("This way it should act as a returning transaction");
    ct2->set_mAnyPointer(this);

    textOut.processTransaction(ct);
    textOut.processTransaction(ct2);

    // Test 4: A LogConfig gets modified and activated. This should be visible in the output.
    textOut.writePure("\nTest 4\nAnother LogConfig was activated. This should be visible in the following output.");
    cfg.reset(false);
    cfg.setAnyPointer(true);
    ct->activateConfig(&cfg);

    textOut.processTransaction(ct);
    ct->deactivateConfig();
  }

protected:
  GC_Logger* pLog;
};

#endif
