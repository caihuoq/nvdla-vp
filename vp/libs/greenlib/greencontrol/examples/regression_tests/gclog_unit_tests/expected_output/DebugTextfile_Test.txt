Tuesday, 16 February 2010 - 13:06:06
Debug text output used with GC_Logger.
Starting log.

User output: This should be the first line in DebugTextfile_Test.txt after the header.
This should be the second line, but without the "User output:" prefix.

Test 2
After this, there should be a single transaction.

  Transaction 0x93a83a0:
    mHasChanged      = 0
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 0
    mSpecifier   (1) = 
    mValue       (1) = 
    mID          (1) = 0x93a1678 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a83e0 (hasChanged: 0)
    -> pNext     (0) = 0

Test 3
Now the same transaction should be printed, followed by another, which acts as a returning CT

  Transaction 0x93a83a0:
    mHasChanged      = 0
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 0
    mSpecifier   (1) = 
    mValue       (1) = 
    mID          (1) = 0x93a1678 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a83e0 (hasChanged: 0)
    -> pNext     (0) = 0
  Transaction 0x93a8418 returning:
    mHasChanged      = 1
    mService     (1) = 2 (AV_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0x93a1678
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 0
    mSpecifier   (1) = This transaction should be modified.
    mValue       (1) = 
    mID          (1) = 0x93a1678 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = This way it should act as a returning transaction
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a8458 (hasChanged: 1)
    -> pNext     (0) = 0

Test 4
Another LogConfig was activated. This should be visible in the following output.

  Transaction 0x93a83a0:
    mHasChanged      = 0
    mService     (0) = 0 (NO_SERVICE)
    mTarget      (0) = 0
    mCmd         (0) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (0) = 0
    mAnyUint     (0) = 0
    mSpecifier   (0) = 
    mValue       (0) = 
    mID          (0) = 0x93a1678 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (0) = 
    mError       (0) = 0
    mLogPointer  (0) = 0
    mCmdIf       (0) = 0
    mLogConfig   (0) = 0xbfe8d8a4 (hasChanged: 1)
    -> pNext     (1) = 0x93a83e0 -> 0

Regular end of log.
Finished on Tuesday, 16 February 2010 - 13:06:06
Closing log.